drivers/interrupt_controller/intc_gicv3.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2020 Broadcom
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+#include <sw_isr_table.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <drivers/interrupt_controller/gic.h>
+
+/* TODO: probe for each implemented rdist base */
+#define GIC_RDIST_BASE  DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1)
+
+#define GICR_SGIBASE_OFFSET	0x10000
```

Comment:
I thought `GICR_` and `GICD_` are as per TRM naming convention and they are short as well.    Rather I would like to see if I have deviated from this anywhere in the driver and change back.

Since `GIC_RDIST_BASE` is going to be a variable with multi core it is better to have `SGI_BASE_OFFSET`  as an independent constant. 

Should I just change as follows.
`GICR_SGIBASE_OFFSET  -> GICR_SGI_BASE_OFFSET`

Reply:
The way Linux handles this is: `GIC_RDIST_BASE` (equivalent) is a macro that retrieves the base address of the local core's redistributor, which I think would be a clean way to handle this.

Having offset definitions is also fine, but we are likely going to have to define separate macros for resolving the actual base address anyway.